src/main/java/games/strategy/engine/config/client/remote/UrlRedirectResolver.java

```diff
@@ -0,0 +1,43 @@
+package games.strategy.engine.config.client.remote;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Module that follows URLs and any redirects and determines the final URL to use.
+ */
+class UrlRedirectResolver {
```

Comment:
This class should probably marked as Deprecated, we should use HTTPComponents instead.
This class does follow a single redirect, but not multiple redirects in a row...
The multiple redirects could be fixed by calling this method recursively when encountering a redirect, but on bad designed webservers this could go infinitely essentially leaading to a StackOverflowError.
If we want to implement everything correctly, so everything is properly handled, we could directly use HttpComponents as this saves us a lot of time!